### DISCUSSION TITLE
Skipping silences like in Newpipe

### DISCUSSION THREAD

**Discussion by manupawickramasinghe** [ReVanced/revanced-discussions#298] [Question] [answered]
I would like an option so the silences are skipped, it would save a lot of time listening to lectures

**Comment by ghost** [comment]
that's a sponserblock specific thing, not a revanced thing, go ask there

**Comment by PacoBell** [comment]
It's not really a SponsorBlock feature. It's an ExoPlayer feature, which I believe the YouTube app relies on. This should technically be possible to enable in ReVanced.

**Comment by andsofine** [comment]
So will it be added or not? 

it is not related to the sponsorblock as it works with any video

**Comment by TheJeterLP** [comment]
> So will it be added or not?
> 
> it is not related to the sponsorblock as it works with any video

As already being said, the devs from sponsorblock need to create this feature so we can implement it. Go and request that feature there.

**Comment by andsofine** [comment]
but it's a player feature

**Comment by TheJeterLP** [comment]
> but it's a player feature

No it's not. Sponsorblock is the way to get the information. We just adapt that feature then.

**Comment by bogadana** [comment, marked as answer]
sponsorblock already has this https://wiki.sponsor.ajay.app/w/Music:_Non-Music_Section

**Reply by suddenlydoomy** [reply]
It's my understanding that SponsorBlock skips silences by crowdsourcing information identifying silences in a video.

What OP is requesting, and what I'm requesting too, is a patch to algorithmically detect and skip silences in a video.  Google Podcasts for Android does this, and so does NewPipe -- not the SponsorBlock fork; original NewPipe.  Since NewPipe is open source, it hopefully wouldn't take too much work.

Does that make sense?

**Reply by oSumAtrIX** [reply]
It would take a considerable amount of work. SponsorBlock does the deed.

**Reply by suddenlydoomy** [reply]
Thank you for the consideration.

**Reply by andsofine** [reply]
> It would take a considerable amount of work. SponsorBlock does the deed.

sponsorblock does a different thing

**Comment by andsofine** [comment]
> > but it's a player feature
> 
> No it's not. Sponsorblock is the way to get the information. We just adapt that feature then. 

in newpipe it works even with videos with only a couple of views. so it can't be because of the sponsorblock

**Reply by TheJeterLP** [reply]
> > > but it's a player feature
> > 
> > No it's not. Sponsorblock is the way to get the information. We just adapt that feature then. 
> 
> in newpipe it works even with videos with only a couple of views. so it can't be because of the sponsorblock
> 

We are not newpipe.

**Reply by oSumAtrIX** [reply]
I mean, it only requires to adjust the integrations to support the sponsor block endpoint, it shouldn't be much effort to simply add it.

**Reply by andsofine** [reply]
> > > > but it's a player feature
> > > 
> > > No it's not. Sponsorblock is the way to get the information. We just adapt that feature then. 
> > 
> > in newpipe it works even with videos with only a couple of views. so it can't be because of the sponsorblock
> > 
> 
> We are not newpipe. 

what is it for?

I say this to explain that the sponsorblock has nothing to do with it

**Comment by LisoUseInAIKyrios** [comment]
https://github.com/ReVanced/revanced-patches/issues/1214
